internal/config/attestation.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+// Package config defines types and interfaces to use for configuring attestation Issuers and Validators.
+package config
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/attestation/measurements"
+	"github.com/edgelesssys/constellation/v2/internal/variant"
+)
+
+// AttestationConfig is the common interface for passing attestation configs.
+type AttestationConfig interface {
+	// GetMeasurements returns the measurements that should be used for attestation.
+	GetMeasurements() measurements.M
+	// GetVariant returns the variant of the attestation config.
+	GetVariant() variant.Variant
+}
+
+// Certificate is a wrapper around x509.Certificate allowing custom marshaling.
+type Certificate x509.Certificate
+
+// MarshalJSON marshals the certificate to PEM.
+func (c Certificate) MarshalJSON() ([]byte, error) {
+	pem := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: c.Raw})
+	return json.Marshal(string(pem))
+}
+
+// MarshalYAML marshals the certificate to PEM.
+func (c Certificate) MarshalYAML() (any, error) {
+	pemData := &bytes.Buffer{}
+	if err := pem.Encode(pemData, &pem.Block{Type: "CERTIFICATE", Bytes: c.Raw}); err != nil {
```

Comment:
Why do we use different funcs for PEM encoding here?

Reply:
You're right, this doesn't make a lot of sense. Will rewrite to just use `pem.EncodeToMemory` instead